src/client.ts

```diff
@@ -393,4 +397,232 @@ export class UXDClient {
+  ): TransactionInstruction {
+    const redeemableAmountUnderManagementCapBN = uiToNative(
+      redeemableAmountUnderManagementCap,
+      controller.redeemableMintDecimals
+    );
+
+    return this.instruction.registerCredixLpDepository(
+      mintingFeeInBps,
+      redeemingFeeInBps,
+      redeemableAmountUnderManagementCapBN,
+      {
+        accounts: {
+          authority,
+          payer: payer ?? authority,
+          controller: controller.pda,
+          depository: depository.pda,
+          collateralMint: depository.collateralMint,
+          depositoryCollateral: depository.depositoryCollateral,
+          depositoryShares: depository.depositoryShares,
+          credixProgramState: depository.credixProgramState,
+          credixGlobalMarketState: depository.credixGlobalMarketState,
+          credixSigningAuthority: depository.credixSigningAuthority,
+          credixLiquidityCollateral: depository.credixLiquidityCollateral,
+          credixSharesMint: depository.credixSharesMint,
+          systemProgram: SystemProgram.programId,
+          tokenProgram: TOKEN_PROGRAM_ID,
+          associatedTokenProgram: ASSOCIATED_TOKEN_PROGRAM_ID,
+          rent: SYSVAR_RENT_PUBKEY,
+        },
+        options,
+      }
+    );
+  }
+
+  public createMintWithCredixLpDepositoryInstruction(
+    controller: Controller,
+    depository: CredixLpDepository,
+    user: PublicKey,
+    collateralAmount: number,
+    options: ConfirmOptions,
+    payer?: PublicKey
+  ): TransactionInstruction {
+    const nativeCollateralAmount = uiToNative(
+      collateralAmount,
+      depository.collateralDecimals
+    );
+
+    const collateralMint = depository.collateralMint;
+    const redeemableMint = controller.redeemableMintPda;
+
+    const [[userCollateral], [userRedeemable]] = findMultipleATAAddSync(user, [
+      collateralMint,
+      redeemableMint,
+    ]);
+
+    return this.instruction.mintWithCredixLpDepository(nativeCollateralAmount, {
+      accounts: {
+        user: user,
+        payer: payer ?? user,
+        controller: controller.pda,
+        depository: depository.pda,
+        depositoryCollateral: depository.depositoryCollateral,
+        depositoryShares: depository.depositoryShares,
+        redeemableMint: redeemableMint,
+        userRedeemable: userRedeemable,
+        collateralMint: collateralMint,
+        userCollateral: userCollateral,
+        credixGlobalMarketState: depository.credixGlobalMarketState,
+        credixSigningAuthority: depository.credixSigningAuthority,
+        credixLiquidityCollateral: depository.credixLiquidityCollateral,
+        credixSharesMint: depository.credixSharesMint,
+        credixPass: depository.credixPass,
+        systemProgram: SystemProgram.programId,
+        tokenProgram: TOKEN_PROGRAM_ID,
+        associatedTokenProgram: ASSOCIATED_TOKEN_PROGRAM_ID,
+        credixProgram: depository.credixProgramId,
+        rent: SYSVAR_RENT_PUBKEY,
+      },
+      options,
+    });
```

Comment:
nit, alternative syntax:

```suggestion
   const {
     depositoryCollateral,
     depositoryShares,
     credixGlobalMarketState,
     credixSigningAuthority,
     credixLiquidityCollateral,
     credixSharesMint,
     credixPass,
     credixProgramId: credixProgram,
   } = depository;

    return this.instruction.mintWithCredixLpDepository(nativeCollateralAmount, {
      accounts: {
        user,
        payer: payer ?? user,
        controller: controller.pda,
        depository: depository.pda,
        depositoryCollateral,
        depositoryShares,
        redeemableMint,
        userRedeemable,
        collateralMint,
        userCollateral,
        credixGlobalMarketState,
        credixSigningAuthority,
        credixLiquidityCollateral,
        credixSharesMint,
        credixPass,
        systemProgram: SystemProgram.programId,
        tokenProgram: TOKEN_PROGRAM_ID,
        associatedTokenProgram: ASSOCIATED_TOKEN_PROGRAM_ID,
        credixProgram,
        rent: SYSVAR_RENT_PUBKEY,
      },
      options,
    });
```

Reply:
Yeah make sense, i personally think that "simpler is better" and using the "dot" syntax is super vanilla and every programmer can understand it, I see a readability reduction when using more advanced syntaxes like so

Reply:
This topic is very opiniated :) Feel free to adapt as you see fit